api/prometheus/v1/api.go

```diff
@@ -139,6 +165,38 @@ type SnapshotResult struct {
 	Name string `json:"name"`
 }
 
+// RulesResult contains the result from querying the rules endpoint.
+type RulesResult struct {
+	Groups []RuleGroup `json:"groups"`
+}
+
+type RuleGroup struct {
```

Comment:
Why not use https://github.com/prometheus/prometheus/blob/master/web/api/v1/api.go#L718 ?

Reply:
The primary reasons for not not importing `RuleGroup` from the `v1` API package referenced above:
-`RuleGroup` models the list of rules as `interface{}` - this requires the client to use type assertion on a private type to determine whether a given rule is a recording or alerting rule.
- It adds an external dependency that the client will need to import in order to use this API client package but I think we can encapsulate the types to mitigate this concern.